feat: add shield rewards modal

lionellbriones opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
- adds reward points details on shield plan
- adds shield reward points modal

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/38379?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: adds shield rewards modal

## **Related issues**

Fixes:

## **Manual testing steps**

1. Login to MetaMask with an account without shield subscription
2. Go to Menu > Settings : click on Transaction Shield
3. Shows Shield Entry modal: click on get started
4. Redirects to Shield plan form
5. Observe what you get list
6. Click on Rewards detail
7. Shows Rewards modal

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**
<img width="372" height="454" alt="Screenshot 2025-12-04 at 12 12 22 AM" src="https://github.com/user-attachments/assets/846898e2-7d1c-4aef-9caf-86cf7a4ace9c" />

<img width="579" height="211" alt="Screenshot 2025-12-04 at 12 13 09 AM" src="https://github.com/user-attachments/assets/842b4c3d-e535-468b-8259-a697a1105698" />



<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Integrates Rewards into Transaction Shield: fetches/ displays estimated points, adds Rewards badge and modal on Shield Plan, and shows rewards info in settings with opt-in.
> 
> - **UI**:
>   - Shield Plan (`ui/pages/shield-plan/shield-plan.tsx`): shows Rewards badge with estimated points per selected interval; adds `ShieldRewardsModal`.
>   - Transaction Shield settings (`ui/pages/settings/transaction-shield-tab/transaction-shield.tsx`): displays rewards info row with estimated points and a Sign up button; loads Rewards onboarding modal; minor icon/color tweaks.
> - **Hooks**:
>   - New `useShieldRewards` in `ui/hooks/subscription/useSubscription.ts` to estimate points (monthly/yearly), check rewards season, and account opt-in.
> - **E2E Mocks**:
>   - Add Rewards API base URL and endpoints; mock responses and handlers (`test/e2e/helpers/shield/constants.ts`, `mocks.ts`).
> - **Tests**:
>   - Initialize rewards duck state in `transaction-shield.test.tsx`.
> - **i18n**:
>   - Add strings for rewards labels, descriptions, and CTA in `app/_locales/en*/messages.json`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 21f4e1c750d6967a9b53424b69d619847219802d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

lionellbriones committed:
feat: add shield rewards modal

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

lionellbriones committed:
feat: update texts

lionellbriones committed:
feat: get season status

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

lionellbriones committed:
feat: check season date

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

lionellbriones committed:
feat: fix lint

lionellbriones committed:
feat: update missed locales

lionellbriones committed:
feat: added rewards uat to privacy snapshot

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

chaitanyapotti commented on ui/pages/shield-plan/shield-plan.tsx:
```diff
@@ -343,7 +350,16 @@ const ShieldPlan = () => {
 
+  const planDetailsRewardsText = useMemo(() => {
+    const points =
+      selectedPlan === RECURRING_INTERVALS.year ? '10,000' : '1,000';
```
shouldn't this come from subscription pricing api? @tanguyenvn

lionellbriones replied:
uses rewardsShield hook now https://github.com/MetaMask/metamask-extension/pull/38379/files#diff-9d5f8d40f5ff16a52171342e5c42a4f3672a168cee24f16d217dac6d050c5d47R107

lionellbriones committed:
feat: typed the dispatch no need to assert seasonMetadata

lionellbriones commented on ui/hooks/rewards/useSeasonStatus.ts:
```diff
@@ -116,3 +117,34 @@ export const useSeasonStatus = ({
+
+  const { value: isRewardsSeason, pending } =
+    useAsyncResult<boolean>(async () => {
+      const seasonMetadata = (await dispatch(
```
updated here https://github.com/MetaMask/metamask-extension/pull/38379/commits/48c38a64c951abc3f9cc02512e5b7f275ff78bc4

lionellbriones committed:
feat: allow closing modal on click outside and escape key

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

lionellbriones committed:
fetch points from rewards

lionellbriones committed:
combine points hook

lionellbriones committed:
return null if there is no selected account

lwin-kyaw commented on ui/hooks/subscription/useSubscription.ts:
```diff
@@ -735,3 +739,65 @@ export const useUpdateSubscriptionCryptoPaymentMethod = ({
+      monthly: monthlyPointsData?.pointsEstimate ?? null,
+      yearly: yearlyPointsData?.pointsEstimate ?? null,
+    };
+  }, [dispatch, accounts, selectedAccount]);
```
This is valid IMO, we can change to [this](https://github.com/MetaMask/metamask-extension/pull/38379/files#r2587762312).

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

lionellbriones commented on ui/pages/shield-plan/shield-plan.tsx:
```diff
@@ -85,19 +87,34 @@ import {
+    pointsMonthly,
+    pointsYearly,
+    pending: pendingPoints,
+  } = useFetchShieldRewardsPoints();
```
makes sense.. will do

lionellbriones committed:
move rewards season inside useSubscription useShieldRewards

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

lionellbriones commented on ui/hooks/subscription/useSubscription.ts:
```diff
@@ -735,3 +739,65 @@ export const useUpdateSubscriptionCryptoPaymentMethod = ({
+
+export const useFetchShieldRewardsPoints = () => {
+  const dispatch = useDispatch<MetaMaskReduxDispatch>();
+  const accounts = useSelector(
```
addressed here https://github.com/MetaMask/metamask-extension/pull/38379/commits/da0cbe729ece5ccb45b065bfb5f4767df3ae8611

lionellbriones committed:
handle useShieldRewards errors

lionellbriones committed:
feat: mock rewards apis for shield e2e

lionellbriones committed:
Updated rewards modal copy

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

lionellbriones committed:
get shield reward details using primary keyring

lionellbriones committed:
Merge branch 'main' into feat/shield-rewards

chaitanyapotti committed:
Merge branch 'main' into feat/shield-rewards

lwin-kyaw committed:
Merge branch 'main' into feat/shield-rewards

lionellbriones committed:
feat: add rewards sign-up on settings (#38527)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
Add rewards sign-up on settings

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/38527?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: add rewards sign-up on settings

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**
<img width="415" height="617" alt="Screenshot 2025-12-04 at 12 29 06 AM"
src="https://github.com/user-attachments/assets/283eba96-513f-48e1-b21f-033f04f0652c"
/>

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds Rewards season points display and a sign-up CTA to Transaction
Shield settings, powered by a new rewards hook and localized strings.
> 
> - **Settings UI (Transaction Shield)**:
> - Show estimated Rewards points (monthly/yearly) during active Rewards
season, formatted by locale.
> - Add “Sign up” CTA when the account hasn’t opted in; opens
`RewardsOnboardingModal`.
> - Integrate `pendingShieldRewards` into loaders; minor icon updates
and color constant fix.
> - **Hooks**:
> - Extend `useShieldRewards` to fetch `hasAccountOptedIn` via
`getRewardsHasAccountOptedIn`, include in return shape, and factor into
pending/error states.
> - **i18n**:
> - Add `shieldTxDetails3Title`, `shieldTxDetails3Description`, and
`shieldTxDetails3DescriptionSignUp` strings to `app/_locales/en*.json`.
> - **Tests**:
> - Update Transaction Shield tests to seed `rewards` initial state and
mock new components.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
4c21ab2281578e112a855ae511ff0653c89e9e4b. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

---------

Co-authored-by: Chaitanya Potti <chaitanya.potti@gmail.com>

lionellbriones committed:
update rewards modal description text
